build/common.ps1

```diff
@@ -240,22 +240,8 @@ Function Install-DotnetCLI {
 
         #If "-force" is specified, or folder with specific version doesn't exist, the download command will run"
         if ($Force -or -not (Test-Path $probeDotnetPath)) {
-            $channelMainVersion = ""
```

Comment:
Doesn't this logic kill our ability to download the latest version from a channel?

Reply:
According to [doc](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script#:~:text=%2DVersion%7C%2D%2Dversion%20%3CVERSION%3E) if we pass version then both `channel` and `-Quality Daily` are ignored. So passing `channel` doesn't have any value if we're passing version. Instead we can opt to remove version here, and keep `channel` + `Quality Daily`.

Reply:
We want to be testing with channel only. 

Tying the version is something we do, but not something we prefer.

Reply:
I see you made a change not using the version right now.

I'm not suggesting we make functional changes, which is what we're doing now. I don't think we should do that. 
We should retain the ability to do both, a version or a channel.

Reply:
How we do that? Only 1 is valid at the time. How about read env var to decide?

Reply:
Sorry I just noticed this. I also worked on this issue when working on https://github.com/NuGet/Client.Engineering/issues/1271, as the SDK version downloaded from Windows and Linux are different if this issue is not fixed.
Here are the changes might be helpful for reference: ( I tested with both channel or a version, they both work)
[scripts/funcTests/runFuncTests.sh](https://github.com/NuGet/NuGet.Client/compare/dev...dev-hengliu-updateSDKForTestingTo6-useVS2022Pool#diff-290139a5aeb3d6be333f8c728c8b6720cc5802e65eb222a47840eeff298a0fc5)
[build/common.ps1](https://github.com/NuGet/NuGet.Client/compare/dev...dev-hengliu-updateSDKForTestingTo6-useVS2022Pool#diff-75700092c78637e43e3336ed5002de2c6d9b3f5e67f517ccdb4a8aba14cb2b25)
[build/config.props](https://github.com/NuGet/NuGet.Client/compare/dev...dev-hengliu-updateSDKForTestingTo6-useVS2022Pool#diff-bd4ef9ba7459727e37706ab7197bdfaa3868ec01297f168ff0b6e243f087d239)

Reply:
BTW, I abandoned lots of logic in build/common.ps1, which is based on the following considerations:
The code is for checking the existing SDK versions in /cli folder before downloading. If the predicted version is already there, it won't download the version again. 
1. The code is not reliable as dotnet installer team might changed the content of URL, or change the URL. We will be in a bad state if there is any change from them.
e.g. the two URLs have different contents, which breaks the script:
https://dotnetcli.blob.core.windows.net/dotnet/Sdk/3.0/latest.version
(two lines, version in the second line)
https://dotnetcli.blob.core.windows.net/dotnet/Sdk/5.0/latest.version)
(only one line, version is the first line)
2. The gain of keeping the code is very small. 
On CI, we will run a clean build every build so the code is useless. 
When we run ./configure.ps1 locally, we usually start with a git clean -xdf command, which makes the code useless, too.
3. The code increases lots of complexity, it doesn't worth when the gain is small.